src/coreclr/src/jit/simdashwintrinsic.cpp

```diff
@@ -0,0 +1,1031 @@
+                        hwIntrinsic = isVectorT256 ? NI_VectorT256_LessThan : NI_VectorT128_LessThan;
+                        op1 = impSimdAsHWIntrinsicRelOp(hwIntrinsic, clsHnd, retType, baseType, simdSize, op1, tmp);
+
+                        // tmp = Zero - op1Dup1
+                        tmp         = gtNewSIMDVectorZero(retType, baseType, simdSize);
+                        hwIntrinsic = isVectorT256 ? NI_AVX2_Subtract : NI_SSE2_Subtract;
+                        tmp = gtNewSimdAsHWIntrinsicNode(retType, tmp, op1Dup1, hwIntrinsic, baseType, simdSize);
+
+                        // result = ConditionalSelect(op1, tmp, op1Dup2)
+                        return impSimdAsHWIntrinsicCndSel(clsHnd, retType, baseType, simdSize, op1, tmp, op1Dup2);
+                    }
+                    break;
+                }
+#elif defined(TARGET_ARM64)
+                case NI_VectorT128_Abs:
+                {
+                    assert(varTypeIsUnsigned(baseType));
+                    return op1;
+                }
+#else
+#error Unsupported platform
+#endif // !TARGET_XARCH && !TARGET_ARM64
+
+                default:
+                {
+                    // Some platforms warn about unhandled switch cases
+                    // We handle it more generally via the assert and nullptr return below.
+                    break;
+                }
+            }
+            break;
+        }
+
+        case 2:
+        {
+            CORINFO_ARG_LIST_HANDLE arg2 = info.compCompHnd->getArgNext(argList);
+            argType                      = JITtype2varType(strip(info.compCompHnd->getArgType(sig, arg2, &argClass)));
+            op2                          = getArgForHWIntrinsic(argType, argClass);
+
+            argType = JITtype2varType(strip(info.compCompHnd->getArgType(sig, argList, &argClass)));
+            op1     = getArgForHWIntrinsic(argType, argClass, isInstanceMethod);
+
+            assert(!SimdAsHWIntrinsicInfo::NeedsOperandsSwapped(intrinsic));
+
+            switch (intrinsic)
+            {
+#if defined(TARGET_XARCH)
+                case NI_Vector2_op_Division:
+                case NI_Vector3_op_Division:
+                {
+                    // Vector2/3 div: since the top-most elements will be zero, we end up
+                    // perfoming 0/0 which is a NAN. Therefore, post division we need to set the
+                    // top-most elements to zero. This is achieved by left logical shift followed
+                    // by right logical shift of the result.
+
+                    // These are 16 byte operations, so we subtract from 16 bytes, not the vector register length.
+                    unsigned shiftCount = 16 - simdSize;
+                    assert((shiftCount > 0) && (shiftCount <= 16));
+
+                    // retNode = Sse.Divide(op1, op2);
+                    GenTree* retNode = gtNewSimdAsHWIntrinsicNode(retType, op1, op2, NI_SSE_Divide, baseType, simdSize);
```

Comment:
Zeroing the unused elements was done in codegen for `GT_SIMD` while we are creating proper nodes to represent the operation here.